### PR TITLE
feat: Implement ReadVolatile and WriteVolatile for TcpStream

### PR DESCRIPTION
**Pull request by ljrcore**
Add implementations of the ReadVolatile and WriteVolatile traits for std::net::TcpStream by extending the impl_read_write_volatile_for_raw_fd! macro.

### CODE REVIEW

**Review by roypat — COMMENTED** [review]
Thanks! Could you also add a changelog entry about this new implementation? :o

**Comment by ljrcore** [comment]
> Thanks! Could you also add a changelog entry about this new implementation? :o

Done.

Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 - \[[#287](https://github.com/rust-vmm/vm-memory/pull/287)\] Support for RISC-V 64-bit platform.
 - \[[#299](https://github.com/rust-vmm/vm-memory/pull/299)\] atomic_bitmap: support enlarging the bitmap.
+- \[[#304](https://github.com/rust-vmm/vm-memory/pull/304)\] Implement ReadVolatile and WriteVolatile for TcpStream
```
Comment (roypat) [inline CHANGELOG.md:11]: Can you move this to the `Upcoming version` section? We already released 0.16 :(

Comment (ljrcore) [reply CHANGELOG.md]: > Can you move this to the `Upcoming version` section? We already released 0.16 :(
> 
> Should be
> 
> ```
> ## Upcoming version
> 
> ### Added
> 
> - \[[#304](https://github.com/rust-vmm/vm-memory/pull/304)\] Implement ReadVolatile and WriteVolatile for TcpStream
> ```

Sorry. I have made the changes accordingly.

Thanks for your detailed correction:)

**Review by roypat — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (roypat) [reply CHANGELOG.md]: Thank you!

**Review by rbradford — APPROVED** [review]

**Pull request merged**